python/plugins/db_manager/dlg_sql_window.py

```diff
@@ -91,10 +91,8 @@ def __init__(self, iface, db, parent=None):
         self.updatePresetsCombobox()
 
         self.geomCombo.setEditable(True)
-        self.geomCombo.lineEdit().setReadOnly(True)
 
         self.uniqueCombo.setEditable(True)
-        self.uniqueCombo.lineEdit().setReadOnly(True)
```

Comment:
> The virtual layer provider is not always able to detect all the column names.

I am almost sure this will lead to serious problems :disappointed: 
> the main UI is the DB manager plugin. I still have to cleanup some code (and UI) for a smoother integration into the main app

:+1:, so this modification of the DB Manager will not be necessary then at all.